pkg/remotewrite/remotewrite.go

```diff
@@ -99,7 +108,12 @@ func (o *Output) Stop() error {
 	defer o.logger.Debug("Output stopped")
 	o.periodicFlusher.Stop()
 
-	staleMarkers := o.staleMarkers(time.Now())
+	// Add 1ms so in the extreme case
+	// that the time frame between the last
+	// and the current flush operations is under-millisecond then
+	// we can avoid that the flush is seen as duplicate keeping the accuracy in the future.
+	timestamp := time.Now().Truncate(time.Millisecond).Add(1 * time.Millisecond)
+	staleMarkers := o.staleMarkers(timestamp)
```

Comment:
I can't say I understand the need and possible side-effects of this, but you'd know better. :smile:

Would it be possible to add a test for it, though? I guess if we're dealing with sub-1ms variations it would be extremely flaky, but have you run into this issue in real world tests? A fix like this should probably be done and discussed in a separate PR.

Reply:
The concept is that the Output (same as k6 is nanosecond based) instead Prometheus is millisecond-based. So during the conversion we lose the accuracy.

>but have you run into this issue in real-world tests

By design it is possible, in the operation `periodFlusher.Stop` it sends "the last batch" then the code arrives here and closes with the stale marker, if this two ops are done in a single millisecond then we will hit a clash and the stale marker will be discarded. "The last batch" is still a remote call so I don't expect in the real world we will hit the issue.